### PR TITLE
Rails 6.1 support

### LINKED ISSUE #26: Resolve rails dependency to 6.1.0
Thank you for the awesome gem!

Currently this gem has a restriction to `< 6.1`, could it be updated to allow using the latest Rails version?

### PR DESCRIPTION
**Pull request by vladimirtemnikov**
Resolves #26

### CODE REVIEW

**Comment by vladimirtemnikov** [comment]
@grosser as I see now in Rails `transaction` method is being called on `model.connection`, not on `model.class`, so I moved some methods to `ActiveRecord::ConnectionAdapters::AbstractAdapter` and removed test case which checks that connection pool does not checkout new connection if there is no current connection.

I also fixed some style warnings and updated old gemfiles and their gemfile.lock.
Do you think it needs to be moved to separate commit or it should be removed?

If you have any suggestions, I'm glad to hear :)

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require:
+  - rubocop-performance
+  - rubocop-rails
+  - rubocop-rspec
+
+AllCops:
+  NewCops: enable
+  TargetRubyVersion: 2.4
```
Comment (grosser) [inline .rubocop.yml:8]: 2.5 is the lowest supported version
```suggestion
  TargetRubyVersion: 2.5 # lowest supported version
```

File: lib/ar_after_transaction.rb [hunk 2]
```diff
@@ -1,72 +1,31 @@
+# frozen_string_literal: true
+
+require 'pry'
```
Comment (grosser) [inline lib/ar_after_transaction.rb:3]:
```suggestion
```

File: lib/ar_after_transaction/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module ARAfterTransaction
```
Comment (grosser) [inline lib/ar_after_transaction/version.rb:1]: should add frozen string comment here ?

File: .reek.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+detectors:
```
Comment (grosser) [inline .reek.yml:1]: need to add reek to dev dependencies and run it as part of CI ?

**Review by grosser — COMMENTED** [review]
usually best to do rubocop/reek changes in it's own PR since it adds lots of noise/bikeshedding, but changes look good overall

**Comment by grosser** [comment]
good enough, I can do the small cleanups 👍 thx so much!

File: .reek.yml (hunk 4 above)
Comment (vladimirtemnikov) [reply .reek.yml:1]: It could be nice addition :)

File: lib/ar_after_transaction/version.rb (hunk 3 above)
Comment (vladimirtemnikov) [reply lib/ar_after_transaction/version.rb:1]: Yeah, I can create new PR with small fixes

**Pull request merged**